opm/common/quad.hpp

```diff
@@ -0,0 +1,325 @@
+};
+
+// provide some type traits for the quadruple precision type
+template <>
+struct is_floating_point<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_arithmetic<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_fundamental<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_scalar<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_pod<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_signed<quad>
+    : public integral_constant<bool, true>
+{};
+
+
+template <>
+struct is_standard_layout<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_trivial<quad>
+    : public integral_constant<bool, true>
+{};
+
+/*
+template <>
+struct is_trivially_copyable<quad>
+    : public integral_constant<bool, true>
+{};
+*/
+
+template <class OtherType>
+struct is_assignable<quad, OtherType>
+    : public integral_constant<bool, is_arithmetic<OtherType>::value>
+{};
+
+template <class OtherType>
+struct is_nothrow_assignable<quad, OtherType>
+    : public is_assignable<quad, OtherType>
+{};
+
+/*
+template <class OtherType>
+struct is_trivially_assignable<quad, OtherType>
+    : public integral_constant<bool, is_arithmetic<OtherType>::value>
+{};
+*/
+
+template <>
+struct is_copy_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_copy_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_move_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_move_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+/*
+template <>
+struct is_trivially_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+*/
+
+template <>
+struct is_copy_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_move_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_move_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+
+template <>
+struct is_destructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_destructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <class OtherType>
+struct is_convertible<quad, OtherType>
+    : public is_arithmetic<OtherType>
+{ };
+
+inline std::ostream& operator<<(std::ostream& os, const quad& val)
+{ return (os << double(val)); }
```

Comment:
yeah, but this is not optimal but it just affects terminal output (which limits precision to 6 decimal digits by default anyway, IIRC). better solutions are appreciated, though.

as you can see, the "raw" libquadmath from GCC is quite rough...

> If not I think it may be better to simply remove this (depends on the amount of trouble that causes...)

this would cause a conpiler error when trying to output any (__float128) scalar value with `std::cout`. I think that is much worse than reduced output precision?